modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -91,6 +92,29 @@ private Guavate() {
 
   //-------------------------------------------------------------------------
   /**
+   * Reducer used in a stream to ensure there is no more than one matching element.
+   * <p>
+   * This method returns an operator that can be used with {@link Stream#reduce(BinaryOperator)}
+   * that returns either zero or one elements from the stream. Unlike {@link Stream#findFirst()}
+   * or {@link Stream#findAny()}, this approach ensures an exception is thrown if there
+   * is more than one element in the stream.
+   * <p>
+   * This would be used as follows:
+   * <pre>
+   *   stream.filter(...).reduce(Guavate.ensureOnlyOne()).get();
+   * </pre>
+   * 
+   * @param <T>  the type of element in the stream
+   * @return the operator
+   */
+  public static <T> BinaryOperator<T> ensureOnlyOne() {
+    return (a, b) -> {
+      throw new IllegalArgumentException(Messages.format("Duplicates found: {} and {}", a, b));
```

Comment:
The message is a bit misleading. `a` and `b` aren't really duplicates. Something like "Multiple values found:"?